submission-history/src/index.ts

```diff
@@ -0,0 +1,281 @@
+// Required libs
+import express from 'express';
+import mysql from 'promise-mysql';
+var fs = require('fs');
+var multer = require('multer');
+var path = require("path");
+const bodyParser = require("body-parser");
+const server = express();
+var SqlString = require('sqlstring'); // To prevent SQL injection
+
+// Use libs
+server.use(express.json());
+server.use(express.urlencoded({ extended: true })); 
+server.use(bodyParser.text()); // for req.body
+server.use(bodyParser.urlencoded({ extended: true })); // for req.body
+// Using EJS as view engine
+server.set('view engine', 'ejs');
+
+// Needed Variables
+// Gerry
+let challengeID = [];
+let challengeName = [];
+let dateSubmittedG = [];
+let didAllTestsPass = [];
+
+// Austin
+let userName = [];
+let userId = [];
+let dateSubmittedA = [];
+
+// Constructor
+async function start() {
+
+    // Create connection
+    const db = await mysql.createConnection({
+        host: process.env.MYSQL_HOST,
+        user: 'root',
+        password: process.env.MYSQL_ROOT_PASSWORD,
+        database: process.env.MYSQL_DATABASE
+    });
+
+    // Create submission
+    server.post('/createSubmission', async (req, res) => {
+        try {
+            var data = {
+                userID: req.body.userID,
+                userName: req.body.userName,
+                challengeId: req.body.challengeId,
+                challengeName: req.body.challengeName,
+                programmingLanguage: req.body.programmingLanguage,
+                dateSubmitted: new Date().toISOString().slice(0, 19).replace('T', ' '),
+                executionTime: req.body.executionTime,
+                didAllTestsPass: req.body.didAllTestsPass
+            };
+            let sql = SqlString.format('INSERT INTO submissionTable SET ?', data);
+            let result = await db.query(sql);
+            res.status(200).json({
+                status: "success",
+                message: "Data inserted..."
+            });
+        }
+        catch (error) {
+            console.log("Insert Submimssion Table " + error);
+            res.status(401).json({
+                status: "failed",
+                message: error
+            });
+        }
+    });
+
+    // Gerry's request
+    // most recent first
+    server.get('/getUserSubmissions', (req, res) => {
+        let sql = SqlString.format('SELECT * FROM submissionTable WHERE userId = ? ORDER BY dateSubmitted DESC', [req.query.userId]);
```

Comment:
You should set a limit for the number of results so you don't need to slice the returned array later.

Reply:
Set the Limit to 10